src/clients/c/tb_client/echo_client.zig

```diff
@@ -62,32 +68,70 @@ pub fn EchoClient(comptime StateMachine_: type, comptime MessageBus: type) type
             self.reply();
         }
 
-        pub fn request(
+        pub fn batch_get(
             self: *Self,
-            user_data: u128,
-            callback: Self.Request.Callback,
             operation: Self.StateMachine.Operation,
-            message: *Message,
-            message_body_size: usize,
-        ) void {
-            const message_request = message.build(.request);
+            event_count: usize,
+        ) Self.BatchError!Batch {
+            const body_size = switch (operation) {
+                inline else => |op| @sizeOf(Self.StateMachine.Event(op)) * event_count,
+            };
+            assert(body_size <= constants.message_body_size_max);
+
+            const demux = self.demux_iops.acquire() orelse return error.TooManyOutstanding;
+            errdefer self.demux_iops.release(demux);
+
+            if (self.messages_available == 0) return error.TooManyOutstanding;
+            const message = self.get_message();
+            errdefer self.release(message);
 
+            const message_request = message.build(.request);
             message_request.header.* = .{
-                .client = 0,
+                .client = self.id,
                 .request = 0,
-                .cluster = 0,
+                .cluster = self.cluster,
                 .command = .request,
                 .operation = vsr.Operation.from(Self.StateMachine, operation),
-                .size = @as(u32, @intCast(@sizeOf(Header) + message_body_size)),
+                .size = @intCast(@sizeOf(Header) + body_size),
             };
 
-            assert(!self.request_queue.full());
+            return Batch{
+                .message = message_request,
+                .event_count = @intCast(event_count),
+                .demux = demux,
+            };
+        }
```

Comment:
So to check my understanding, the echo client's `batch_get`/`batch_submit` don't actually do any batching, right? i.e. each "batch_get" ends up in a separate request message no matter what.

But then why does `Batch` have a `demux` property at all?